corehq/apps/userreports/models.py

```diff
@@ -778,7 +785,53 @@ def ui_filters(self):
     @property
     @memoized
     def charts(self):
-        return [ChartFactory.from_spec(g._obj) for g in self.configured_charts]
+        if (
+            self.config_id and self.configured_charts
+            and toggles.SUPPORT_EXPANDED_COLUMN_IN_REPORTS.enabled(self.domain)
+        ):
+            configured_charts = deepcopy(self.configured_charts)
+            for chart in configured_charts:
+                if chart['type'] == 'multibar':
+                    chart['y_axis_columns'] = self._get_expanded_y_axis_cols_for_multibar(chart['y_axis_columns'])
+            return [ChartFactory.from_spec(g._obj) for g in configured_charts]
+        else:
+            return [ChartFactory.from_spec(g._obj) for g in self.configured_charts]
+
+    def _get_expanded_y_axis_cols_for_multibar(self, original_y_axis_columns):
+        y_axis_columns = []
+        try:
+            for y_axis_column in original_y_axis_columns:
+                if isinstance(y_axis_column, dict):
```

Comment:
Do you know how these two differences come to exist and whether we can just prevent that, rather than having to catch it everywhere it occurs. If not it might be nice to move this into a `get_column_id` util, so that it can be reused in other places.

Reply:
I think its always meant to be a dict but some tests have this set as a string but I dont understand reports fully to confirm that. I did see the report that I had created on the UI and it was a dict.

Reply:
removed this if-else now in https://github.com/dimagi/commcare-hq/pull/30873/commits/03da0e3ab765d59c1dcd110f68985748e6f34cd3 and used only dict.

I could confirm from code at least that this is always expected to be a dict from other usages.
Wrapping for configured charts happening [here](https://github.com/dimagi/commcare-hq/blob/9b6bd74d8167daffcc4b2c2b2e659a00cbb774ee/corehq/apps/userreports/models.py#L796) that expects spec for chart to be
https://github.com/dimagi/commcare-hq/blob/9c880402f4315060f8cec97939733b50d6868ecc/corehq/apps/userreports/reports/specs.py#L579-L584


The tests that were failing earlier are not failing now since this was put behind a feature flag.